### PR TITLE
Discard empty stack frames

### LINKED ISSUE #1570: Native Java stacktraces from PlatformExceptions have a trailing empty stack frame
### Platform

Flutter Mobile

### Obfuscation

Disabled

### Debug Info

Disabled

### Doctor

-

### Version

7.4.2

### Steps to Reproduce

1. Get a native Java stacktrace via PlatformException
2. Report it to Sentry

### Expected Result

No trailing empty stack frame

### Actual Result

Screenshot 2023-07-26 at 08 09 57

### Are you willing to submit a PR?

None

### PR DESCRIPTION
**Pull request by denrase**
## :scroll: Description

Filter out stack frames that are empty.

## :bulb: Motivation and Context

Closes #1570

## :green_heart: How did you test it?

Unit tests. Check on sentry.io

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

**Comment by denrase** [comment]
@ueman Would you be so kind to check if this solves your issue?

**Comment by ueman** [comment]
@denrase I'll try to get to it tomorrow 👍

**Comment by ueman** [comment]
The problematic `PlatformException` looks like this when printed to console. Unfortunately, I can't reproduce it on my devices, but I found this in the breadcrumbs. Interestingly enough, there's no empty frame, so I assume there's still something else wrong with the current algorithm.
```
PlatformException(error, Unsupported value: '[Ljava.lang.StackTraceElement;@42f1d34' of type 'class [Ljava.lang.StackTraceElement;', null, java.lang.IllegalArgumentException: Unsupported value: '[Ljava.lang.StackTraceElement;@42f1d34' of type 'class [Ljava.lang.StackTraceElement;'
	at m9.q.p(StandardMessageCodec.java:67)
	at m9.r.c(StandardMethodCodec.java:7)
	at m9.j$a$a.error(MethodChannel.java:1)
	at com.dexterous.flutterlocalnotifications.FlutterLocalNotificationsPlugin.getNotificationChannels(FlutterLocalNotificationsPlugin.java:9)
	at com.dexterous.flutterlocalnotifications.FlutterLocalNotificationsPlugin.onMethodCall(FlutterLocalNotificationsPlugin.java:21)
	at m9.j$a.a(MethodChannel.java:2)
	at z8.c.l(DartMessenger.java:2)
	at z8.c.m(DartMessenger.java:3)
	at z8.c.i(Unknown Source:0)
	at z8.b.run(Unknown Source:12)
	at android.os.Handler.handleCallback(Handler.java:955)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:228)
	at android.app.ActivityThread.main(ActivityThread.java:9105)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:614)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1129)
)
```

For convenience, here's just the Java stacktrace:
```
java.lang.IllegalArgumentException: Unsupported value: '[Ljava.lang.StackTraceElement;@42f1d34' of type 'class [Ljava.lang.StackTraceElement;'
	at m9.q.p(StandardMessageCodec.java:67)
	at m9.r.c(StandardMethodCodec.java:7)
	at m9.j$a$a.error(MethodChannel.java:1)
	at com.dexterous.flutterlocalnotifications.FlutterLocalNotificationsPlugin.getNotificationChannels(FlutterLocalNotificationsPlugin.java:9)
	at com.dexterous.flutterlocalnotifications.FlutterLocalNotificationsPlugin.onMethodCall(FlutterLocalNotificationsPlugin.java:21)
	at m9.j$a.a(MethodChannel.java:2)
	at z8.c.l(DartMessenger.java:2)
	at z8.c.m(DartMessenger.java:3)
	at z8.c.i(Unknown Source:0)
	at z8.b.run(Unknown Source:12)
	at android.os.Handler.handleCallback(Handler.java:955)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:228)
	at android.app.ActivityThread.main(ActivityThread.java:9105)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:614)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1129)
```

Maybe there's even just an issue with a trailing newline? The closing `)` of the exception is on itss own line, which could be an indicator for that? Comparing the stacktrace to the screenshot of the issues seems to indicate that there's nothing missing, so the empty line seems to be superflouos.

File: flutter/lib/src/jvm/jvm_exception.dart [hunk 1]
```diff
@@ -256,8 +256,11 @@ class JvmException {
       frames.add(trimmed);
     }
 
-    final thisExceptionFrames =
-        thisException.map((e) => JvmFrame.parse(e)).toList(growable: false);
+    final thisExceptionFrames = thisException
+        .map((e) => JvmFrame.parse(e))
+        .where(
+            (e) => e.originalFrame.isNotEmpty && e.originalFrame.trim() != 'at')
+        .toList(growable: false);
```
Comment (ueman) [inline flutter/lib/src/jvm/jvm_exception.dart]:
```suggestion
    final thisExceptionFrames = thisException
        .where((line) => line.trim().isNotEmpty)
        .map((e) => JvmFrame.parse(e))
        .toList(growable: false);
```

File: flutter/test/jvm/jvm_exception_test.dart [hunk 2]
```diff
@@ -194,3 +204,21 @@ java.lang.IllegalArgumentException: Unsupported value: '[Ljava.lang.StackTraceEl
...
+    at android.os.Handler.handleCallback(Handler.java:942)
+    at android.os.Handler.dispatchMessage(Handler.java:99)
+    at android.os.Looper.loopOnce(Looper.java:201)
+    at android.os.Looper.loop(Looper.java:288)
+    at android.app.ActivityThread.main(ActivityThread.java:7872)
+    at java.lang.reflect.Method.invoke
+    at com.android.internal.os.RuntimeInit\$MethodAndArgsCaller.run(RuntimeInit.java:548)
+    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
+    at
+    ''';
```
Comment (ueman) [inline flutter/test/jvm/jvm_exception_test.dart]:
```suggestion

    ''';
```

File: flutter/lib/src/jvm/jvm_exception.dart [hunk 3]
```diff
@@ -256,8 +256,10 @@ class JvmException {
       frames.add(trimmed);
     }
 
-    final thisExceptionFrames =
-        thisException.map((e) => JvmFrame.parse(e)).toList(growable: false);
+    final thisExceptionFrames = thisException
+        .where((line) => line.trim().isNotEmpty)
```
Comment (ueman) [inline flutter/lib/src/jvm/jvm_exception.dart:262]: Maybe explain with a comment why it's being done like that?
```suggestion
        // Sometimes stringified exceptions from the native side have
        // empty lines. Discard those!
        .where((line) => line.trim().isNotEmpty)
```

**Review by ueman — APPROVED** [review]
Thank you for fixing this. LGTM 🎉

**Review by buenaflor — APPROVED** [review]

**Pull request merged**